src/vue/forms/AtomicSwapQuoteAssetsForm.vue

```diff
@@ -0,0 +1,292 @@
+  selectedSameAssetCode,
+  cardNumber,
+  address,
+} from '@validators'
+import { mapGetters } from 'vuex'
+import { vuexTypes } from '@/vuex'
+import { globalize } from '@/vue/filters/globalize'
+
+const EVENTS = {
+  submit: 'submit',
+}
+
+export default {
+  name: 'atomic-swap-quote-assets-form',
+  mixins: [FormMixin],
+  props: {
+    isDisabled: { type: Boolean, default: false },
+  },
+  data: _ => ({
+    form: {
+      quoteAssets: [{ price: '', asset: {}, destination: '' }],
+    },
+    isLoaded: false,
+    isLoadFailed: false,
+    isFormSubmitting: false,
+  }),
+
+  validations () {
+    return {
+      form: {
+        quoteAssets: {
+          $each: {
+            asset: {
+              required,
+              selectedSameAssetCode: asset => {
+                /* eslint-disable max-len */
+                const selectedAssets = this.getSelectedAssetsByAssetCode(asset.code)
+                return selectedSameAssetCode(selectedAssets)
+                /* eslint-enable max-len */
+              },
+            },
+            destination: {
+              required,
+              addressOrNumberCard: (value, quoteAsset) => {
+                return quoteAsset.asset.isCoinpayments
+                  ? address(quoteAsset.asset.code)(value)
+                  : cardNumber(value)
+              },
+            },
+          },
+        },
+      },
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      quoteAtomicSwapAssets: vuexTypes.quoteAtomicSwapAssets,
+    }),
+  },
+
+  async created () {
+    this.form.quoteAssets[0].asset = this.quoteAtomicSwapAssets[0] || {}
+  },
+  methods: {
+    submit () {
+      this.$emit(EVENTS.submit, this.form)
+    },
+
+    getSelectedAssetsByAssetCode (assetCode) {
+      const selectedAssetsCode = []
+      this.form.quoteAssets.forEach(quoteAsset => {
+        if (quoteAsset.asset.code === assetCode) {
+          selectedAssetsCode.push(quoteAsset.asset.code)
+        }
+      })
+      return selectedAssetsCode
+    },
+
+    setQuoteAssetByCode (code, index) {
+      this.form.quoteAssets[index].asset = this.quoteAtomicSwapAssets
+        .find(item => item.code === code)
+    },
+
+    addQuoteAsset () {
+      this.form.quoteAssets.push({
+        destination: '',
+        asset: this.quoteAtomicSwapAssets[0],
+      })
+    },
+
+    deleteQuoteAsset (index) {
+      this.form.quoteAssets.splice(index, 1)
+    },
+
+    canAddQuoteAsset (index) {
+      return index === this.form.quoteAssets.length
+    },
+
+    canDeleteQuoteAsset (index) {
+      return index !== this.form.quoteAssets.length ||
+        (index === this.form.quoteAssets.length &&
+          this.form.quoteAssets.length !== 1)
+    },
+
+    getDestinationLabel (asset) {
+      return globalize(
+        asset.isCoinpayments
+          ? 'atomic-swap-quote-assets-form.address-lbl'
+          : 'atomic-swap-quote-assets-form.card-number-lbl'
+      )
+    },
+  },
+}
+</script>
+
+<style lang="scss" scoped>
```

Comment:
remove word `quote`
example `add-quote-asset` to `add-asset`
in `atomic-swap-quote-assets-form` located only quote assets

Reply:
renamed